index.ts

```diff
@@ -229,45 +262,48 @@ export function bulk ({
  * @param {string} [$0.url]                     optional URL to override ESRI reverseGeocode endpoint
  * @return {Promise}                            A Promise that'll get resolved with reverse geocode result
  */
-export function reverse ({
+export function reverse({
   clientId,
   clientSecret,
   forStorage = false,
   point,
   url
-}: BaseQuery & {
-  forStorage?: boolean,
-  point: any
-}): Promise<BaseResponse> {
+}: BaseQuery &
+  Options & {
+    point: LonLatInput
+  }): Promise<BaseResponse> {
```

Comment:
Two things:

1. Can you in-line the applicable keys within the `Options` type since not all of those values would apply as keys in the input argument for this function call?
2. Can you also please extend the `BaseResponse` type with the `query` key/value since the query is eventually returned?